libpldm/pdr.c

```diff
@@ -1707,6 +1719,12 @@ uint32_t pldm_entity_association_pdr_add_contained_entity(
 			prev = curr;
 			curr = curr->next;
 		}
+		if (curr == NULL) {
+			printf("Couldn't find bmc record handle %d\n",
+			       bmc_record_handle);
+			free(new_record);
```

Comment:
Can we instead make "added=false" and at line 1774 already free is done based on the "added" flag